app/scripts/controllers/app-metadata.ts

```diff
@@ -0,0 +1,99 @@
+import EventEmitter from 'events';
+import { ObservableStore } from '@metamask/obs-store';
+import ExtensionPlatform from '../platforms/extension';
+
+const platform = new ExtensionPlatform();
+
+/**
+ * The state of the AppMetadataController
+ */
+export type AppMetadataControllerState = {
+  currentAppVersion: string;
+  previousAppVersion: string;
+  previousMigrationVersion: string;
+  currentMigrationVersion: string;
+};
+
+/**
+ * The options that NetworkController takes.
+ */
+export type AppMetadataControllerOptions = {
+  currentAppVersion?: string;
+  state?: {
+    currentAppVersion?: string;
+    previousAppVersion?: string;
+    previousMigrationVersion?: string;
+    currentMigrationVersion?: string;
+  };
```

Comment:
Nit: you can reference the state type here:

```suggestion
  state?: Partial<AppMetadataControllerState>;
```